159: consolidate side panel / bundle detail page

ChengshuLi opened the pull request:
Fixes #159 

@percyliang 

I have taken out the 'fetchExtra' code since I think it's probably not the most efficient way of doing things nor the cleanest code. My assumption is that 'fetchExtra' is for updating run bundle status? I will try to finish the other PR (automatic updating for run bundle) (https://github.com/codalab/codalab-worksheets/pull/146) concurrently so that I won't break things.

klopyrev commented:
Eric, you are dropping the code that was added here: https://github.com/codalab/codalab-worksheets/pull/180

ChengshuLi commented:
@percyliang

percyliang commented:
Comments on the behavior (without having looking at the code):
- When try to edit the description (or any metadata) on the bundle details page, get this in the JS console:
editable_field.js:30 Uncaught TypeError: this.props.onChange is not a function
- Colors should be applied to the editable fields just like they are for the side panel
- Replace 'Bundle Details' with 'Bundle: <bundle name>'
- Make sure you merge the latest develop so that you get the newest set of fields which are displayed

percyliang commented:
@ChengshuLi this is nice - code looks fine, just some issues with functionality to be ironed out (see above)

ChengshuLi commented:
@percyliang

percyliang commented:
@ChengshuLi there is for some reason too much spacing between rows in the metadata in the side panel (wasn't the case before).  Was this deliberate?  Can you please revert the spacing to something closer to what was there before?

percyliang commented:
@ChengshuLi Also, as the last bullet mentioned, the set of metadata fields displayed is not correct (time, data_size) are now at the top.  Please check out develop and make sure that your changes preserve functionality.

ChengshuLi commented:
@percyliang
1) I increased the spacing because the editable field resized the row every time we toggle it. I made the editable field look thinner now and reverted back to the spacing like before (with tiny increase).
2) fixed metadata field
3) I put the "download button" at the last row of metadata and I think it would be clearer for users that this button is used for downloading bundle. I remembered the first time I used Codalab that wasn't obvious to me.

percyliang commented on codalab/apps/web/static/js/worksheet/editable_field.jsx:
```diff
@@ -22,7 +22,7 @@ var EditableField = React.createClass({
           return response.exception;
         }
-        if ('onChange' in this.props) {
+        if (this.props.onChange != undefined) {
```
!==

percyliang commented on codalab/apps/web/templates/web/bundles/detail.html:
```diff
@@ -1,7 +1,7 @@
 
 {% block head_title %}Bundle{% endblock head_title %}
-{% block page_title %}Bundle Details{% endblock page_title %}
+{% block page_title %}Bundle: {% endblock page_title %}
```
Remove ':'

Actually, is there a way to get the the name up here?
If not, can we just remove this header completely?  It doesn't really add much value and takes up space.

percyliang commented:
Overall looks good!  LGTM modulo the 2 comments
